pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -389,6 +389,7 @@ func (a *Adapter) createOrUpdateComponent(
 
 	annotations := make(map[string]string)
 	odolabels.SetProjectType(annotations, component.GetComponentTypeFromDevfileMetadata(a.AdapterContext.Devfile.Data.GetMetadata()))
+	odolabels.AddCommonAnnotations(annotations)
```

Comment:
Would it also make sense to add this annotation to resources created by `odo deploy` as well?
Unless I've missed something, it seems this annotation is only added to resources created by `odo dev`.

Reply:
It seems not possible to create a component from `odo deploy` for now:
https://github.com/redhat-developer/odo/issues/6376#issuecomment-1351407029